packages/fuel-indexer/src/commands/run.rs

```diff
@@ -156,28 +157,18 @@ pub async fn exec(args: IndexerArgs) -> anyhow::Result<()> {
                 addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 4000),
                 ..Config::local_node()
             };
-            tokio::spawn(async move {
-                let node = FuelService::new_node(config).await.unwrap();
-                Some(node)
-            })
-        } else {
-            tokio::spawn(futures::future::ready(None))
+            subsystems.spawn(async move {
+                if let Err(e) = FuelService::new_node(config).await {
+                    tracing::error!("Fuen Node failed: {e}");
+                };
+            });
         }
     };
 
-    #[cfg(not(feature = "fuel-core-lib"))]
-    let node_handle = tokio::spawn(futures::future::ready(()));
-
-    // spawn application as separate task
-    tokio::spawn({
-        let cancel_token = cancel_token.clone();
-        async move {
-            let _ = tokio::join!(service_handle, node_handle, web_handle);
-            cancel_token.cancel();
-        }
-    });
-
-    cancel_token.cancelled().await;
+    // Each subsystem runs its own loop, and we require all subsystems for the
+    // Indexer service to operate correctly. If any of the subsystems stops
+    // running, the entire Indexer Service exits.
+    subsystems.join_next().await;
```

Comment:
If a service stops, does this actually cause the others to exit? Or are we relying on the fact that an error will be returned in the respective components and waiting for `join_next()` to catch it?

I don't mind this, but what do you think about something a bit more explicit?:
```
if let Some(_res) = subsystems.join_next().await {
    subsystems.shutdown().await;
}
```